dbt/include/global_project/macros/adapters/columns.sql

```diff
@@ -123,11 +123,11 @@
      alter {{ relation.type }} {{ relation.render() }}
 
             {% for column in add_columns %}
-               add column {{ column.name }} {{ column.data_type }}{{ ',' if not loop.last }}
+               add column {{ adapter.quote(column.name) }} {{ column.data_type }}{{ ',' if not loop.last }}
```

Comment:
Do we always want to quote? Or is there a config that we need to condition on, similar to the quote policy for relation names?

Reply:
I think adapter.quote leverages the same  quote policy as relation names

Reply:
@mikealfare are you thinking something like [this](https://github.com/dbt-labs/dbt-adapters/blob/782a32b530e25afe34af6896392d9fb081dce9d1/dbt/include/global_project/macros/relations/rename.sql#L31) (for relations) or [this](https://github.com/dbt-labs/dbt-adapters/blob/782a32b530e25afe34af6896392d9fb081dce9d1/dbt/include/global_project/macros/relations/table/create.sql#L45) (for columns)?

In this particular case, I think we need to match the quoting from [here](https://github.com/dbt-labs/dbt-core/blob/5310d3715cdaa109eea03995731425a85cd6cd33/core/dbt/include/global_project/macros/materializations/models/incremental/column_helpers.sql#L11) (which would mean always quoting via `adapter.quote()`). I forget the exact details -- and I could be wrong! -- but there's some notes under the "**Root Cause**" toggle in https://github.com/dbt-labs/dbt-adapters/issues/250#issuecomment-2192228487.

Reply:
Side note: here are the primary resources I've been using when researching quoting or case-sensitive issues:
- https://github.com/dbt-labs/docs.getdbt.com/issues/3518
- https://github.com/dbt-labs/dbt-core/issues/2986

Reply:
@dbeatty10 oh these are really coll gonna dig in haven't seen these before.

Reply:
@colin-rogers-dbt `adapter.quote()` will always [quote](https://github.com/dbt-labs/dbt-adapters/blob/43ea32d821ed041cdc03175308e1f411211847a3/dbt/adapters/sql/impl.py#L209-L210).

@dbeatty10 If the objective is to always quote the column in this case, that's fine. I just know we've run into issues before where we quote something that doesn't get quoted, or vice versa, and then it fails a condition in a filter where it's supposed to pass. I haven't seen it with columns, so maybe it doesn't apply here.